magma/syntax/sequential2.py

```diff
@@ -205,11 +297,30 @@ def sequential2(pre_passes=[], post_passes=[],
                 annotations: Optional[dict] = None,
                 **clock_io_kwargs):
     """ clock_io_kwargs used for ClockIO params, e.g. async_reset """
-    passes = (pre_passes +
-              [ssa(strict=False), bool_to_bit(), if_to_phi(_seq_phi)] +
-              post_passes)
+    passes = (
+        pre_passes +
+        [ReplacePrev(), ssa(strict=False), bool_to_bit(), if_to_phi(_seq_phi)] +
+        post_passes
+    )
+
+    reset_type = None
+    if ("has_async_reset" in clock_io_kwargs and
+            clock_io_kwargs["has_async_reset"]):
+        reset_type = ast.parse("m.AsyncReset").body[0].value
+    elif ("has_async_resetn" in clock_io_kwargs and
+            clock_io_kwargs["has_async_resetn"]):
+        reset_type = ast.parse("m.AsyncResetN").body[0].value
+    elif "has_reset" in clock_io_kwargs and clock_io_kwargs["has_reset"]:
+        reset_type = ast.parse("m.Reset").body[0].value
+    elif "has_resetn" in clock_io_kwargs and clock_io_kwargs["has_resetn"]:
+        reset_type = ast.parse("m.ResetN").body[0].value
```

Comment:
Yes, I think it would be good to update the sequential2 interface to support the reset_type parameter for consistency with the new register interface.